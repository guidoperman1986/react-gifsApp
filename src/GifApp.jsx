import React, { useState } from 'react'
import { AddCategory, GifGrid } from './components';

export const GifApp = () => {
    const [categories, setCategories] = useState([])

    const onAddCategory = (category) => {
        if (categories.find(cat=>cat.toLowerCase() === category.toLowerCase())) return;

        setCategories((categories)=>[category, ...categories])
    }

    const deleteGifs = () => {
        setCategories([])
    }

    return (
        <>
            <div className='title-bar'>
                <h1>GifApp</h1>
                {
                    categories.length > 0 && 
                        <img 
                            onClick={deleteGifs} 
                            width={40} 
                            src="https://icongr.am/fontawesome/close.svg?size=128&color=currentColor" 
                            alt="" 
                        />
                }
            </div>
            
            <AddCategory onNewCategory={ (value) => onAddCategory(value)}></AddCategory>
            {
                categories.map(category => 
                    <GifGrid 
                        key={category} 
                        category={category} 
                    /> 
                )
            }           
        </>
    )
}
