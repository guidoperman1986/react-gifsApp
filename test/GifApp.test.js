import { fireEvent, render, screen } from "@testing-library/react"
import { GifApp } from "../src/GifApp"

describe('Test on GifApp', () => {
    test('should match the snapshot', () => {
        const { container } = render(<GifApp />)
        screen.debug();
        expect(container).toMatchSnapshot()

    })

    test('should show h1 with text GifApp', () => {
        render(<GifApp />)
        const h1 = screen.getAllByText('GifApp');
        expect(h1).toBeTruthy();
    })
    
    test('should have a form', () => {
        render(<GifApp />)
        const form = screen.getAllByRole('form')
        expect(form).toBeTruthy();        
    })
    
    test('should have an input', ()=> {
        render(<GifApp />)
        const input = screen.getByRole('input');
        expect(input).toBeTruthy();    
    })
    
    test('should type into the input', () => {
        render(<GifApp />)
        const input = screen.getByRole('input');
        expect(input.value).toBe('')
        fireEvent.change(input, {target: {value: '123'}});
        expect(input.value).toBe('123')
    })
})