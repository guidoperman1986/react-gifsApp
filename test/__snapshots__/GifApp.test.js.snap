// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on GifApp should match the snapshot 1`] = `
<div>
  <div
    class="title-bar"
  >
    <h1>
      GifApp
    </h1>
  </div>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs..."
      role="input"
      type="text"
      value=""
    />
  </form>
</div>
`;
